internal/scaffold/helm/watches.go

```diff
@@ -43,6 +52,58 @@ func (s *WatchesYAML) GetInput() (input.Input, error) {
 	return s.Input, nil
 }
 
+// UpdateHelmWatchForResource checks for duplicate GVK, and appends to existing Watch.yaml file.
+func UpdateHelmWatchForResource(r *scaffold.Resource, absProjectPath string, chart string) error {
```

Comment:
These updater functions would be easier to unit test/reason about if they had typed parameters and returned either an updated type or `[]byte`, ex:

```go
func UpdateWatchForResource(r *scaffold.Resource, absProjectPath string, chart string) error {
	watchFilePath := filepath.Join(absProjectPath, WatchesYamlFile)
	watchYAML, err := ioutil.ReadFile(watchFilePath)
	if err != nil {
		return fmt.Errorf("failed to read watch manifest %v: %v", watchFilePath, err)
	}
	if len(watchYAML) == 0 {
		return fmt.Errorf("empty Watch File at: %v", absProjectPath)
	}
	watchList := []watches.Watch{}
	err = yaml.Unmarshal(watchYAML, &watchList)
	if err != nil {
		return fmt.Errorf("failed to unmarshal watch config %v ", err)
	}

	watchList, err = addWatchForResource(r, watchList, chart)
 	if err != nil {
		return err
	}

	data, err := yaml.Marshal(watchList)
	if err != nil {
		return fmt.Errorf("failed to marshal watch config: %v", err)
	}

	if err := ioutil.WriteFile(watchFilePath, data, fileutil.DefaultFileMode); err != nil {
		return fmt.Errorf("failed to update %v: %v", watchFilePath, err)
	}
}

func addWatchForResource(r *scaffold.Resource, original []watches.Watch, chart string) []watches.Watch {
	gvk := schema.GroupVersionKind{
		Version: r.Version,
		Group:   r.FullGroup,
		Kind:    r.Kind,
	}

	for _, watch := range original {
		if watch.GroupVersionKind == gvk {
			// dupe detected
			return nil, fmt.Errorf("duplicate GVK: %s", watch.GroupVersionKind)
		}
	}
	if chart == "" {
		chart = r.LowerKind
	}
	newWatch := watches.Watch{
		GroupVersionKind: gvk,
		ChartDir:         HelmChartsDir + "/" + chart,
	}
	return append(original, newWatch), nil
}
```

Reply:
Will take up in a follow up PR, as this needs refactoring the unit tests as well.
@joelanford WDYT?

Reply:
Since you're adding the unit tests here and not modifying existing ones I'd say this change is in-scope for this PR.

Reply:
If the purpose is to make unit testing easier by verifying the returned `watches.yaml` content, Then current unit test cases are already covering it. https://github.com/bharathi-tenneti/operator-sdk/blob/ansible/helm-addapi/internal/scaffold/helm/watches_test.go#L153

Reply:
As per discussion, taking this in for Follow-up.